src/App.jsx

```diff
@@ -3,21 +3,39 @@ import React, { Suspense } from 'react'
 import NavBar from './components/NavBar'
 import Spinner from './components/Spinner'
 import AppContainer from './components/AppContainer'
+import Select from './components/Select'
 import CovInce from './Covince'
 
+import useDarkMode from './hooks/useDarkMode'
+
 const Loading = () => (
   <div className='fixed inset-0 grid place-content-center'>
-    <Spinner className='w-6 h-6 text-gray-500' />
+    <Spinner className='w-6 h-6 text-gray-500 dark:text-gray-300' />
   </div>
 )
 
 function App () {
+  const { isDark, mode, setMode } = useDarkMode()
   return (
     <>
       <Suspense fallback={<Loading />}>
-        <NavBar />
+        <NavBar>
+          <label className='text-sm flex items-center space-x-3'>
+            <span className='font-bold'>Theme:</span>
+            <Select
```

Comment:
This is super cool for dev, and previewing now, but IMO takes up a bit too much attention to stay forever on the `covince`  implementation. But it's probably something we can deal with down the line (finding a subtler version, not for this review) if you want to show off the functionality.